MeshLib/PropertyVector.h

```diff
@@ -12,29 +12,84 @@
 #define PROPERTYVECTOR_H_
 
 #include <algorithm>
+#include <iostream>
+#include <iterator>
 #include <vector>
 #include <type_traits>
 #include <memory>
 
+#include "BaseLib/excludeObjectCopy.h"
+
 namespace MeshLib
 {
 
+template <typename T>
+class PropertyVector;
+
+class PropertyVectorBase
+{
+public:
+	virtual PropertyVectorBase* clone(
+		std::vector<std::size_t> const& exclude_positions
```

Comment:
how about changing here to pass a pointer instead of a reference and setting a default value of nullptr?